test/foundry/unit/concrete/erc4337account/TestERC4337Account_AddDeposit.t.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "../../../utils/Imports.sol";
+import "../../../utils/SmartAccountTestLab.t.sol";
+
+    event DepositAdded(address indexed account, address indexed depositor, uint256 amount);
+
+contract TestERC4337Account_addDeposit is Test, SmartAccountTestLab {
+    SmartAccount private account;
+
+    function setUp() public {
+                init();
+                account = BOB_ACCOUNT;
+    }
+
+    function test_AddDeposit_Success() public {
+        uint balanceBefore = ENTRYPOINT.balanceOf(address(account));
+        uint256 depositAmount = 1 ether;
+
+        account.addDeposit{value: depositAmount}();
+
+        assertEq(balanceBefore + depositAmount, ENTRYPOINT.balanceOf(address(account)), "Deposit should be added to EntryPoint");
+    }
+
+    function test_AddDeposit_EventEmitted() public {
+        uint256 depositAmount = 1 ether;
+        (bool res, ) = address(account).call{value: depositAmount}(""); // Pre-funding the account contract
+        assertTrue(res, "Pre-funding account should succeed");
+
+        vm.expectEmit(true, true, true, true);
+        emit DepositAdded(address(account), address(this), depositAmount); // Assuming there's a DepositAdded event
+
+        account.addDeposit{value: depositAmount }();
+    }
+
+    function test_AddDeposit_Revert_NoValue() public {
+        account.addDeposit();
+    }
+
+function test_AddDeposit_DepositViaHandleOps() public {
+    (bool res, ) = payable(address(BOB_ACCOUNT)).call{ value: 2 ether }(""); // Fund account with 2 ETH
+    assertTrue(res, "Pre-funding account should succeed");
+    uint256 depositAmount = 1 ether;
+    uint balanceBefore = ENTRYPOINT.balanceOf(address(account));
+
+    Execution[] memory execution = new Execution[](1);
+    execution[0] = Execution(address(account), depositAmount, abi.encodeWithSignature("addDeposit()"));
```

Comment:
this case doesn't make sense to do it with user operation? account depositing into entrypoint from native tokens deposited already within itself?
because generally our modifier addMissingFunds takes care of it...

Reply:
It's unit testing, checking all cases, even the obvious, also try to reproduce all case through handleOps flow.